docs/blog/bundling-dependencies.md

```diff
@@ -0,0 +1,185 @@
+- We want to patch a dependency in a way that the dependency is unlikely to ever accept upstream
+
+Disadvantages:
+
+- Packages no longer receive downloads
+  - The number of downloads is a commonly used metric for sponsoring OSS projects, or deciding if to use such a package
+- Dependencies no longer receive updates
+  - They will be locked to the version at the time of bundling. The burden to release security updates is then on you (to re-bundle)
+- Bloated dependency trees are hidden but still exist
+  - These dependencies would be better replaced by alternatives, or contributed to
+- npm de-duplication no longer happens
+  - If a consumer has many copies of this dependency, the bundled ones will be duplicates since npm isn't aware of them
+- Maintainers don't receive help/contributions to improve
+  - Bundling the dependency often means the core problem is put off for longer (e.g. bloated sub-dependencies). Helping the maintainers solve the problem would be better
+- Issues are often opened with your project rather than the dependency
+
+### Tree-shaking in libraries
+
+Bundlers are very good at tree-shaking these days, so you can quite easily pull in a larger library and only use part of it without needing to ship all of it.
+
+However, if you're publishing a library, this means you're placing the job of tree-shaking onto your consumers (assuming you document that they should do it!).
+
+This isn't great, but neither is the alternative (bundling the parts you use).
+
+If possible, you really should try find a more appropriate package or create one which provides the part of the functionality you need.
+
+### Extracting new packages
+
+Sometimes, a package has a piece of functionality you need but that accounts for a tiny amount of the overall size.
+
+If this is the case, it often makes sense to extract it into its own package.
+
+However, the line is _very fine_ here.
+
+If you extract something too little, you're publishing micro-utilities (which we're trying to avoid).
+
+Ideally, you should extract _groups_ of functionality into re-usable packages. A good example of this is [empathic](https://github.com/lukeed/empathic), a group of path utilities.
+
+### De-duplication
+
+Whatever you do, if you bundle, you're at risk of removing the benefits npm gives of de-duplication.
+
+Ideally, we want to declare our dependencies so we can leave npm to de-dupe and share them across other libraries being used in parallel to ours.
+
+## Real-world examples
+
+### vite
+
+vite currently bundles most of its dependencies as a way to tree-shake things up front so its consumers don't have to.
+
+Given that it is a developer tool, unused in production, this made sense at the time it was decided.
+
+However, dependencies have vastly improved since then and the community has been hard at work improving performance across the board. These days, it is likely the vite team would not choose to bundle most of their dependencies.
```

Comment:
This sounds a bit too optimistic. Last time we discussed this option with the team, there was heavy push back. I don't know when are we going to be unbundling vite deps, and if we do it will probably happen gradually.
```suggestion
However, dependencies have vastly improved since then and the community has been hard at work improving performance across the board. It is likely that the vite team will stop bundling some of their dependencies in the future.
```